src/main/scala/openflights.scala

```diff
@@ -147,30 +158,29 @@ object OpenFlights{
     }
   }
 
-  def loadRoutesAsMap(path: String): Map[RouteKey,List[Route]] = {
+  private def loadRoutesAsMap(path: String): Map[RouteKey, List[Route]] = {
     val records = loadCSVDataFile(path)
-    records.foldLeft(Map[RouteKey,List[Route]]()) { (map, n) => 
-      var src      = n(2).replace("\"","")
-      var dst      = n(4).replace("\"","")
-      
+    val index = Map.empty[RouteKey, List[Route]].withDefaultValue(List())
+
+    records.foldLeft(index) { (index, n) =>
+      var src      = n(2).replace("\"", "")
+      var dst      = n(4).replace("\"", "")
+
       // Both src and dst airport codes as a key
-      var routeKey = RouteKey(src,dst)
+      var routeKey = RouteKey(src, dst)
       var route    = Route(
-        n(0).replace("\"",""),
-        n(2).replace("\"",""),
-        n(4).replace("\"",""),
+        n(0).replace("\"", ""),
+        src,
+        dst,
         n(7).toInt
       )
 
-      if (map.contains(routeKey))
-        map(routeKey) = map(routeKey).+:(route)
-      else
-        map(routeKey) = List(route)
-      map
+      index(routeKey) = route +: index(routeKey)
```

Comment:
Oh I meant to mention about the syntax here: it is a built-in part of the language that methods ending in `:` are right-associative, that's why you had to use the awkward syntax to call it as `map(routeKey).+:(route)`.

The documentation suggests the mnemonic for remembering these infix operator methods that *the COLon goes on the side with the COLlection*, e.g.

```scala
route +: index(routeKey)
```

is a prepend operation, while

```scala
index(routeKey) :+ route
```

would be append. Using prepend gets us results in reverse-alphabetical order in the current app that `Core` runs, but it's probably the right choice since prepend is the efficient operation on immutable `List`, even though these are probably going to be very small lists so it isn't a big deal. It's probably best in general to prepend and make a `sort` call just before displaying if that's desired, the result list is usually small.

FWIW I renamed `map` to `index` just because `map` is such a common method name in functional code that it made me squint. Maybe `index` isn't much better since it's similar to a counter when iterating, but I was thinking of it like "secondary index" in database terms, which is basically how these maps are being used…

If the `:` right-associative thing seems like a weird choice, it might help to know that it was driven partly by using `::` as the fundamental functional list cons operator:

```
scala> 3 :: 2 :: 1 :: List(0)
res0: List[Int] = List(3, 2, 1, 0)
```

So that's a synonym for `+:` on `List`, but `+:` is used more broadly on other collection types as well.